WeatherBotService/WeatherBotStation/Data/WeatherDataObservable.cs

```diff
@@ -0,0 +1,32 @@
+using WeatherBotStation.WeatherBots;
+using WeatherBotStation.WeatherBots.BotManager;
+
+namespace WeatherBotStation.Data;
+
+public class WeatherDataObservable(IWeatherBotManager manager) : IWeatherDataObservable
+{
+    private readonly IList<IWeatherBot> _bots = manager.GetBots();
+    private WeatherData _weatherData = null!;
+
+    public WeatherData WeatherData
+    {
+        get => _weatherData;
+        set
+        {
+            _weatherData = value;
+            Notify();
```

Comment:
Won't introducing a function for processing weather data and calling **Notify()** method inside it make more sense and much clearer?

public void Process(WeatherData data)
{
   ....
   Notify() // private method 
}